app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingPage.kt

```diff
@@ -0,0 +1,165 @@
+            .padding(bottom = 16.dp)
+            .then(modifier),
+    ) {
+        val boxWithConstraintsScope = this
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .verticalScroll(rememberScrollState()),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.SpaceBetween,
+        ) {
+            IconButton(
+                onClick = onDismiss,
+                modifier = Modifier.align(Alignment.End),
+            ) {
+                Icon(
+                    painter = painterResource(id = R.drawable.mozac_ic_close),
+                    contentDescription = stringResource(R.string.content_description_close_button),
+                    tint = FirefoxTheme.colors.iconPrimary,
+                )
+            }
+
+            Column(
+                modifier = Modifier.padding(horizontal = 16.dp),
+                horizontalAlignment = Alignment.CenterHorizontally,
+                verticalArrangement = Arrangement.Center,
+            ) {
+                Image(
+                    painter = painterResource(id = pageState.image),
+                    contentDescription = null,
+                    modifier = Modifier
+                        .height(boxWithConstraintsScope.maxHeight.times(IMAGE_HEIGHT_RATIO)),
+                )
+
+                Spacer(modifier = Modifier.height(32.dp))
+
+                Text(
+                    text = pageState.title,
+                    color = FirefoxTheme.colors.textPrimary,
+                    textAlign = TextAlign.Center,
+                    style = FirefoxTheme.typography.headline5,
+                )
+
+                Spacer(modifier = Modifier.height(16.dp))
+
+                Text(
+                    text = pageState.description,
+                    color = FirefoxTheme.colors.textSecondary,
+                    textAlign = TextAlign.Center,
+                    style = FirefoxTheme.typography.body2,
+                )
+            }
+
+            Spacer(modifier = Modifier.height(16.dp))
+
+            Column(
+                horizontalAlignment = Alignment.CenterHorizontally,
+                modifier = Modifier.padding(horizontal = 16.dp),
+            ) {
+                PrimaryButton(
+                    text = pageState.primaryButtonText,
+                    onClick = onPrimaryButtonClick,
+                )
+
+                if (pageState.secondaryButtonText != null) {
+                    Spacer(modifier = Modifier.height(8.dp))
+                    SecondaryButton(
+                        text = pageState.secondaryButtonText,
+                        onClick = onSecondaryButtonClick,
+                    )
+                }
+            }
+
+            LaunchedEffect(pageState) {
+                pageState.onRecordImpressionEvent()
+            }
+        }
+    }
+}
+
+private const val IMAGE_HEIGHT_RATIO = 0.4f
```

Comment:
Let's move this to the top of the file and could you add a comment about how you arrive at this number for others to understand that this isn't just a magic number.